src/ui/router/Browse.js

```diff
@@ -19,10 +19,13 @@ define([
 
         openmct.router.on('change:params', function (newParams, oldParams, changed) {
             if (changed.view && browseObject) {
-                let provider = openmct
-                    .objectViews
-                    .getByProviderKey(changed.view);
-                viewObject(browseObject, provider);
+                //get latest version of the browseObject
+                openmct.objects.get(browseObject.identifier).then((newBrowseObject) => {
```

Comment:
This results in an additional object fetch per path change. The problem here, as I understand it, is that the two route handlers are both being invoked for path changes, but that the `change:params` handler is being invoked first, with the old object. `navigateToPath` will already fetch the new object, so I think the right approach here is to invoke the `change:params` handler at the end of `navigateToPath` and to add a semaphore that is checked in `change:params` that prevents it being called twice.

Reply:
agree to the part where when path changes we call object fetch twice. The issue will still remain when we are only changing the view, which only fires the change:params event. We will still need to get the latest object when only change:params is invoked.

I think the only way to avoid the double fetch would be to have a semaphore when the original fetch happens (when the change:path event fires) to avoid another fetch when the change:params event fires. When only the change:params event fires, we still have to fetch the latest object

Reply:
@deeptailor I see, didn't realize that was an issue as well. How has this not come up before? In any case, `Browse.js` [is already listening for changes to the navigated object](https://github.com/nasa/openmct/blob/87d838c69094d818e1807404b72f8238de86c691/src/ui/router/Browse.js#L64), but right now it doesn't update the `browseObject` variable. If we just updated `browseObject` in the mutation listener as well then I think we're good?